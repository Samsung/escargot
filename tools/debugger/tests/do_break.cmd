break do_break.js:53
b do_break.js:39
b f
c
c
c
c
