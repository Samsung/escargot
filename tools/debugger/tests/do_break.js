/*
 * Copyright (c) 2020-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */


var cat = 'cat';

function test(x)
{
  function f() {
    return 0;
  }

  function f() {
    /* Again. */
    return 1;
  }

  function f() {
    /* And again. */
    return 2;
  }

  var a = 3;
  var b = 5, c = a + b;
  global_var = f();
  return c;
}

function f() {
  /* And again. */
}

var
  x =
    1;

test(x);
