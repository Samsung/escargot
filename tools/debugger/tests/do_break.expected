Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_break.js:21
(escargot-debugger) break do_break.js:53
Breakpoint 1 at tools/debugger/tests/do_break.js:53
(escargot-debugger) b do_break.js:39
Breakpoint 2 at tools/debugger/tests/do_break.js:39 (in test() at line:23, col:1)
(escargot-debugger) b f
Breakpoint 3 at tools/debugger/tests/do_break.js:26 (in f() at line:25, col:3)
Breakpoint 4 at tools/debugger/tests/do_break.js:31 (in f() at line:29, col:3)
Breakpoint 5 at tools/debugger/tests/do_break.js:36 (in f() at line:34, col:3)
Breakpoint 6 at tools/debugger/tests/do_break.js:47 (in f() at line:45, col:1)
(escargot-debugger) c
Stopped at breakpoint:1 tools/debugger/tests/do_break.js:53
(escargot-debugger) c
Stopped at breakpoint:2 tools/debugger/tests/do_break.js:39 (in test() at line:23, col:1)
(escargot-debugger) c
Stopped at breakpoint:5 tools/debugger/tests/do_break.js:36 (in f() at line:34, col:3)
(escargot-debugger) c
Connection closed.
